instagram-app/src/components/PostContainer/PostContainer.js

```diff
@@ -0,0 +1,72 @@
+import React, {useState} from 'react'
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import uuid from 'uuid';
+import './PostContainer.css'
+import {Input, InputGroup, InputGroupAddon, Button} from 'reactstrap'
+import CommentSection from '../CommentSection/CommentSection'
+const PostContainer = ({data})=>{
+    const imgStyle = {
+        width: '100%'
+    }
+    const time = ()=>{
+        const timestamp = data.timestamp.split('');
+        const timestamps= timestamp.filter(date=>!(date.includes('t')||date.includes('h')));
+        const date = Date.parse(timestamps.join(''))
+       return(moment(date).endOf('day').fromNow())
+    }
+
+    const [comment, addComment] = useState(data.comments)
+    const [message, sendMessage] = useState('')
+    return (
+        <div className='PostContainer'>
+            <div className='posts'>
```

Comment:
This component it's working fine but it's a mess. It could be splitted in additional children components, the anonymous function you are using should me moved as methods of the class etc...

Try to work on this to have more short, easy to read, reusable components instead of huge chaotic one!